src/freenet/clients/http/utils/UriFilterProxyHeaderParser.java

```diff
@@ -0,0 +1,99 @@
+package freenet.clients.http.utils;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import freenet.config.Option;
+import freenet.support.MultiValueTable;
+
+public class UriFilterProxyHeaderParser {
+    private final Option<?> fProxyPortConfig;
+    private final Option<?> fProxyBindToConfig;
+    private final String uriScheme;
+    private final String uriHost;
+    private final MultiValueTable<String, String> headers;
+
+    public UriFilterProxyHeaderParser(
+        Option<?> fProxyPort,
+        Option<?> fProxyBindTo,
+        String uriScheme,
+        String uriHost,
+        MultiValueTable<String, String> headers
+        ) {
+
+        this.fProxyPortConfig = fProxyPort;
+        this.fProxyBindToConfig = fProxyBindTo;
+        this.uriScheme = uriScheme;
+        this.uriHost = uriHost;
+        this.headers = headers;
+    }
+
+    public static SchemeAndHostWithPort parse (
+        Option<?> fProxyPort,
+        Option<?> fProxyBindTo,
+        String uriScheme,
+        String uriHost,
+        MultiValueTable<String, String> headers
+        ) {
+        return new UriFilterProxyHeaderParser(fProxyPort, fProxyBindTo, uriScheme, uriHost, headers)
+            .retrieveSchemeHostAndPort();
+    }
+
+    public class SchemeAndHostWithPort {
+        private final String scheme;
+        private final String host;
+
+        SchemeAndHostWithPort(
+            String scheme,
+            String host
+        ) {
+            this.scheme = scheme;
+            this.host = host;
+        }
+
+        public String toString() {
```

Comment:
Hmm. Yes, this does what’s advertised but I’m hesitant to have “real” business logic in a method that was originally meant for debugging. However, feel free to let it this way, it doesn’t feel important enough. :)